### PR TITLE
Transfer module

### PR DESCRIPTION
**Pull request by sayfullah-aleph-lab**
-Pulled from master branch 
- rectified custom-http issue 
- added Koin dependency
- added ui for transfer module

### CODE REVIEW

File: app/src/main/java/com/moneytransfer/demo/MainActivity.kt [hunk 1]
```diff
@@ -7,6 +7,7 @@ import androidx.navigation.findNavController
 import androidx.navigation.ui.AppBarConfiguration
 import androidx.navigation.ui.setupActionBarWithNavController
 import androidx.navigation.ui.setupWithNavController
+import com.moneytransfer.core.SampleCore
```
Comment (SalmanAshraf70) [inline app/src/main/java/com/moneytransfer/demo/MainActivity.kt]: remove this, we are not using this import

File: app/build.gradle [hunk 2]
```diff
@@ -52,4 +62,6 @@ dependencies {
     testImplementation 'junit:junit:4.13.2'
     androidTestImplementation 'androidx.test.ext:junit:1.1.3'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+
+
```
Comment (SalmanAshraf70) [inline app/build.gradle:78]: avoid extra spacing

File: app/src/main/java/com/moneytransfer/demo/MainActivity.kt [hunk 3]
```diff
@@ -31,5 +32,6 @@ class MainActivity : AppCompatActivity() {
         )
         setupActionBarWithNavController(navController, appBarConfiguration)
         navView.setupWithNavController(navController)
+
```
Comment (SalmanAshraf70) [inline app/src/main/java/com/moneytransfer/demo/MainActivity.kt:34]: remove this space line too

File: app/src/main/java/com/moneytransfer/demo/MainApplication.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+//            modules(
+//                listOf(
+//                    repositoryModule,
+//                    useCaseModule,
+//                    viewModelModule
+//                )
+//            )
+//        }
+
+
```
Comment (SalmanAshraf70) [inline app/src/main/java/com/moneytransfer/demo/MainApplication.kt]: remove all commented code and extra empty lines

File: app/src/main/java/com/moneytransfer/demo/di/ViewModelModule.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.moneytransfer.demo.ui.transfer.TransferViewModel
+import java.lang.reflect.Array.get
+
+
+
+
+//val applicationModule = module {
+//    single { AppRepository }
+//}
```
Comment (SalmanAshraf70) [inline app/src/main/java/com/moneytransfer/demo/di/ViewModelModule.kt]: remove commented code

File: app/src/main/java/com/moneytransfer/demo/MainActivity.kt (hunk 1 above)
Comment (sayfullah-aleph-lab) [reply app/src/main/java/com/moneytransfer/demo/MainActivity.kt]: Noted.

File: app/build.gradle (hunk 2 above)
Comment (sayfullah-aleph-lab) [reply app/build.gradle:78]: noted

File: app/src/main/java/com/moneytransfer/demo/MainActivity.kt (hunk 3 above)
Comment (sayfullah-aleph-lab) [reply app/src/main/java/com/moneytransfer/demo/MainActivity.kt:34]: I'll remove the changes from the app module and add these on the transfer module

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -28,6 +28,6 @@
         app:layout_constraintLeft_toLeftOf="parent"
         app:layout_constraintRight_toRightOf="parent"
         app:layout_constraintTop_toTopOf="parent"
-        app:navGraph="@navigation/mobile_navigation" />
+        app:navGraph="@navigation/transfer_navigation" />
```
Comment (SalmanAshraf70) [inline app/src/main/res/layout/activity_main.xml]: why you change the navigation file name here. it should be inside the transfer module not here.

**Pull request closed without merging**